cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/CXXAmbiguitiesTest.kt

```diff
@@ -57,8 +57,10 @@ class CXXAmbiguitiesTest {
             }
         assertNotNull(tu)
 
-        // make sure we still have only one declaration in the file (the record)
-        assertEquals(1, tu.declarations.size)
+        // we have 3 (record) declarations in our TU now. 1 of the original MyClass and two because
+        // CDT thinks that "call" is the return type and "crazy" the type of the parameter. We infer
+        // record declarations for all types, so we end up with 3 declarations here.
```

Comment:
And we haven't inferred the record declarations until now? Do we actually have those types in the final graph too? Shouldn't we replace these types since we don't really know?

Reply:
Yes exactly, we did not infer record declaration of types until now. I am not sure if there is anything more I can do at this point 🤷